python/cuspatial/cuspatial/tests/binpreds/test_binpred_test_dispatch.py

```diff
@@ -71,75 +61,103 @@ def test_simple_features(
         The pytest request object. Used to print the test name in
         diagnostic output.
     """
-    try:
+    if not LOG_DISPATCHED_PREDICATES:
         (lhs, rhs) = simple_test[2], simple_test[3]
         gpdlhs = lhs.to_geopandas()
         gpdrhs = rhs.to_geopandas()
+
+        # Reverse
+        pred_fn = getattr(rhs, predicate)
+        got = pred_fn(lhs)
+        gpd_pred_fn = getattr(gpdrhs, predicate)
+        expected = gpd_pred_fn(gpdlhs)
+        assert (got.values_host == expected.values).all()
+
+        # Forward
         pred_fn = getattr(lhs, predicate)
         got = pred_fn(rhs)
         gpd_pred_fn = getattr(gpdlhs, predicate)
         expected = gpd_pred_fn(gpdrhs)
         assert (got.values_host == expected.values).all()
-
-        # The test is complete, the rest is just logging.
+    else:
         try:
-            # The test passed, store the results.
-            predicate_passes[predicate] = (
+            (lhs, rhs) = simple_test[2], simple_test[3]
+            gpdlhs = lhs.to_geopandas()
+            gpdrhs = rhs.to_geopandas()
+
+            # Reverse
+            pred_fn = getattr(rhs, predicate)
+            got = pred_fn(lhs)
+            gpd_pred_fn = getattr(gpdrhs, predicate)
+            expected = gpd_pred_fn(gpdlhs)
+            assert (got.values_host == expected.values).all()
+
+            # Forward
+            pred_fn = getattr(lhs, predicate)
+            got = pred_fn(rhs)
+            gpd_pred_fn = getattr(gpdlhs, predicate)
+            expected = gpd_pred_fn(gpdrhs)
+            assert (got.values_host == expected.values).all()
```

Comment:
Can we wrap this in a helper? Let's strive to avoid code duplication. Esp. this part is the actual test running portion right? It's significant enough to be singled out.

Reply:
done